### PR TITLE
[4.0] Sticky table header

### PR DESCRIPTION
**Pull request by brianteeman**
This pr leverages a fixed bug in chrome 91 to allow the setting of position sticky to table headers.

To test rebuild the css `npm run build:css`

On any admin view containing a table eg list of article or list of menu items make sure that you have more rows in the table than the height of your browser.

Scroll down and the toolbar is sticky but the table headers disappear
Apply this PR
Scroll down and the toolbar is sticky and the table headers are sticky

If your browser does not support this then its just ignored and no change will be observed.

### CODE REVIEW

**Comment by Quy** [comment]
Go to Home Dashboard.

**Comment by brianteeman** [comment]
I can fix that easily enough - if this PR is desired.

**Comment by Quy** [comment]
On mobile.
34593-mobile

I am for this PR.

**Comment by brianteeman** [comment]
easy fixes. thanks for testing them. I will fix both tonight

File: administrator/templates/atum/scss/vendor/bootstrap/_table.scss [hunk 1]
```diff
@@ -84,5 +84,15 @@
 
   .j-main-container > & {
     box-shadow: 0 2px 10px -8px var(--atum-bg-dark-50);
+
+    thead {
+      position: sticky;
+      top: calc(43px + 16px + .5rem); // min-height + padding + margin-bottom
+      box-shadow: 0 2px 10px -8px var(--atum-bg-dark-50);
+
+      @media (max-width: 575.98px) {
```
Comment (Quy) [inline administrator/templates/atum/scss/vendor/bootstrap/_table.scss]:
```suggestion
      @include media-breakpoint-down(sm) {
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 021a45d7c96e2297bac1e8c47355b65dfadad663<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34593">issues.joomla.org/tracker/joomla-cms/34593</a>.</sub>

**Comment by Quy** [comment]
Please fix conflict.

Please fix the vertical ellipse in the header under Categories.

**Comment by richard67** [comment]
@brianteeman Could you solve the conflict here in `_table.scss`? I think it comes from PR #34621 , so you should change `var(--atum-bg-dark-50)` to `var(--template-bg-dark-50)`.

**Comment by brianteeman** [comment]
conflicts resolved

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 4c1757db9ef51bcf56ebd8c80ab3cb321f10ccb1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34593">issues.joomla.org/tracker/joomla-cms/34593</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 4c1757db9ef51bcf56ebd8c80ab3cb321f10ccb1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34593">issues.joomla.org/tracker/joomla-cms/34593</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34593">issues.joomla.org/tracker/joomla-cms/34593</a>.</sub>

**Comment by bembelimen** [comment]
I like this idea a lot, just two minor thoughts:
- I'm not sure about the box-shadow TBH, it gives a 3d effect which does not fit to 100% to the flat table (I see, that there is a "separator" needed when scrolling)
- Is it possible to prevent that it's glued to the navbar?

grafik

**Comment by bembelimen** [comment]
@brianteeman could you please check my two ~~issues~~ points? If there is an improvement possible (especially with the last one) I would be happy, otherwise I will merge.

**Comment by brianteeman** [comment]
> I'm not sure about the box-shadow TBH, it gives a 3d effect which does not fit to 100% to the flat table (I see, that there is a "separator" needed when scrolling)

The box shadow you refer to exists **before** this pr

> Is it possible to prevent that it's glued to the navbar?

Not sure I understand you

**Comment by bembelimen** [comment]
> > I'm not sure about the box-shadow TBH, it gives a 3d effect which does not fit to 100% to the flat table (I see, that there is a "separator" needed when scrolling)
> 
> The box shadow you refer to exists **before** this pr

Before:
grafik


With PR:
grafik

This is the shadow I mean: https://github.com/joomla/joomla-cms/pull/34593/files#diff-d59aacea1f8715ae7515cde80c9ae4919762203616cd7f05d95aa4653d0b41fbR87

> 
> > Is it possible to prevent that it's glued to the navbar?
> 
> Not sure I understand you

grafik

Here they're not glued together, have a bit of spacing

**Comment by richard67** [comment]
@brianteeman Could you check @bembelimen 's previous comment? Thanks in advance.

**Comment by wilsonge** [comment]
Back to pending as effectively there's a undesired change in the th styling found by @bembelimen (should be a quick fix I imagine)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34593">issues.joomla.org/tracker/joomla-cms/34593</a>.</sub>

**Comment by richard67** [comment]
Setting status "Needs Review" in the issue tracker.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34593">issues.joomla.org/tracker/joomla-cms/34593</a>.</sub>

**Pull request closed without merging**